frog/imports/ui/Preview/ShowDashExample.js

```diff
@@ -52,58 +64,215 @@ class ShowDashExample extends React.Component<PropsT, StateT> {
+      logs = suppliedLogs || this.state.logs.slice(this.state.oldSlider, e);
+    } else {
+      tempDb = pureObjectReactive(
+        cloneDeep(aT.dashboard[this.state.example].initData)
+      );
+      logs = this.state.logs.slice(0, e);
+    }
+    const [doc, dataFn] = tempDb;
+    this.activityDbObject.actualClosingTime =
+      (suppliedLogs && suppliedLogs[suppliedLogs.length - 1].timestamp) ||
+      this.state.logs[e].timestamp;
+    logs.forEach(log => mergeLog(doc.data, dataFn, log, this.activityDbObject));
+
+    this.setState({ data: tempDb[0].data, oldSlider: e });
   };
 
+  throttledDisplaySubset = throttle(this.displaySubset, 500, {
+    leading: false
+  });
+
   render() {
-    // const instances = this.props.activityType.dashboard.exampleLogs[
-    //   this.props.example
-    // ].instances || []
     const instances = [];
     const users = {};
     const { logs } = this.state;
+    const aT = this.props.activityType;
+    const dashNames = Object.keys(aT.dashboard).filter(
+      x => aT.dashboard[x].exampleLogs
+    );
+    const examples = aT.dashboard[this.state.example].exampleLogs.map(
+      x => x.title
+    );
+    const Viewer = aT.dashboard[this.state.example].Viewer;
+    if (!this.state.logs) {
+      return <Spinner />;
+    }
     return (
       <React.Fragment>
+        <DashboardSelector
+          selected={this.state.exampleIdx}
+          onChange={x => {
+            this.logsProcessed = 0;
+            this.setState(
+              {
+                oldSlider: 0,
+                data: aT.dashboard[dashNames[x]].initData,
+                example: dashNames[x],
+                exampleIdx: x,
+                logs: [],
+                idx: 0,
+                play: false
+              },
+              () => this.fetchLogs()
+            );
+          }}
+          dashNames={dashNames}
+        />
+        <DashboardSelector
+          selected={this.state.idx}
+          onChange={x => {
+            this.logsProcessed = 0;
+            this.setState(
+              {
+                oldSlider: 0,
+                data: aT.dashboard[this.state.example].initData,
+                logs: [],
+                idx: x,
+                play: false
+              },
+              () => this.fetchLogs()
+            );
+          }}
+          dashNames={examples}
+        />
+        <DashboardSelector
+          onChange={x => {
+            if (this.state.play === x) {
+              this.setState({ play: false });
+            } else {
+              this.setState(
+                {
+                  oldSlider: 0,
+                  data: aT.dashboard[this.state.example].initData,
+                  play: x
+                },
+                () => {
+                  this.logsProcessed = 0;
+                  this.clusterBySecond();
+                }
+              );
+            }
+          }}
+          dashNames={['1x', '2x', '4x', '8x', '16x', '32x']}
```

Comment:
UI-wise the "x1 x2 x4" appBar makes no sense as we are not changing tabs. Why not making that a button on the left of the slider (a dropdown for example that lets you chose the speed and then click "Play"). That will maybe also fix the bugs about switching speed. I know we will clean up the UI later, but this seems like a quick win.